### PR TITLE
Ch14 - Accurate and Consistent

### PR DESCRIPTION
**Pull request by treygrainger**

### CODE REVIEW

File: chapters/ch12/0.setup.ipynb [hunk 1]
```diff
@@ -116,10 +116,37 @@
...
       "\n",
-      "Successfully written 48204 documents\n"
+      "Successfully written 48194 documents\n"
+     ]
+    },
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "----------------------------------------\n",
```
Comment (treygrainger) [inline chapters/ch12/0.setup.ipynb:129]: @dcrouch26 - output in this notebook has errors. Please re-run to generate clean output (and ensure it matches book output).

File: chapters/ch14/4.question-answering-demo-application.ipynb [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "ename": "OSError",
+     "evalue": "Can't load the configuration of '../../data/roberta-base-squad2-outdoors'. If you were trying to load it from 'https://huggingface.co/models', make sure you don't have a local directory with the same name. Otherwise, make sure '../../data/roberta-base-squad2-outdoors' is the correct path to a directory containing a config.json file",
```
Comment (treygrainger) [inline chapters/ch14/4.question-answering-demo-application.ipynb:196]: @dcrouch26 output has errors. Please rerun to ensure these are working and their outputs are consistent with the book.

**Comment by treygrainger** [comment]
This PR used to have `ch12` as the base, but looks like yesterday you changed the base branch to main and force pushed. Currently the PR view is showing that most (all?) the `ch12` branch changes are now duplicated in this PR view. Should we just merge the two together?

**Comment by treygrainger** [comment]
Nevermind the last comment. I just changed the base branch back to 12 and it looks fine now.

**Comment by dcrouch26** [comment]
Merging into 12 for verification

File: engines/solr/SolrCollection.py [hunk 3]
```diff
@@ -115,6 +115,31 @@ def transform_response(self, search_response):
         
     def native_search(self, request=None, data=None):
         return requests.post(f"{SOLR_URL}/{self.name}/select", json=request, data=data).json()
+    
+    def vector_search(self, **search_args):
```
[the remark this answers is not in the document]
Comment (dcrouch26) [reply engines/solr/SolrCollection.py]: Seems this didn't get merged in

**Pull request merged**